crates/bevy_anti_alias/src/fsr3/mod.rs

```diff
@@ -0,0 +1,549 @@
+    prepass::{DepthPrepass, MotionVectorPrepass, ViewPrepassTextures},
+};
+use bevy_diagnostic::FrameCount;
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    query::{QueryItem, With},
+    reflect::ReflectComponent,
+    schedule::IntoScheduleConfigs,
+    system::{Commands, Query, Res, ResMut},
+    world::World,
+};
+use bevy_image::ToExtents;
+use bevy_math::{UVec2, Vec2, Vec4Swizzles};
+use bevy_reflect::{std_traits::ReflectDefault, Reflect, ReflectFromReflect};
+use bevy_render::{
+    camera::{ExtractedCamera, MipBias, TemporalJitter},
+    render_graph::{NodeRunError, RenderGraphContext, RenderGraphExt, ViewNode, ViewNodeRunner},
+    render_resource::{
+        Buffer, BufferDescriptor, BufferUsages, TextureDescriptor, TextureDimension, TextureFormat,
+        TextureUsages,
+    },
+    renderer::{RenderDevice, RenderQueue},
+    sync_component::SyncComponentPlugin,
+    sync_world::RenderEntity,
+    texture::{CachedTexture, TextureCache},
+    view::{ExtractedView, Hdr, Msaa, ViewTarget},
+    ExtractSchedule, MainWorld, Render, RenderApp, RenderSystems,
+};
+use bevy_time::Time;
+use std::{mem::size_of, sync::Mutex};
+use tracing::warn;
+use wgpu_ffx::{FsrContext, FsrContextFlags, FsrContextInfo, FsrDispatchFlags, FsrDispatchInfo};
+
+/// Plugin for AMD FidelityFX Super Resolution 3.
+///
+/// See [`Fsr3`] for more details.
+#[derive(Default)]
+pub struct Fsr3Plugin;
+
+impl Plugin for Fsr3Plugin {
+    fn build(&self, app: &mut App) {
+        app.add_plugins(SyncComponentPlugin::<Fsr3>::default());
+
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app
+            .add_systems(ExtractSchedule, extract_fsr3_settings)
+            .add_systems(
+                Render,
+                (
+                    prepare_fsr3_jitter_and_context.in_set(RenderSystems::ManageViews),
+                    prepare_fsr3_textures.in_set(RenderSystems::PrepareResources),
+                ),
+            )
+            .add_render_graph_node::<ViewNodeRunner<Fsr3Node>>(Core3d, Node3d::Fsr3)
+            .add_render_graph_edges(
+                Core3d,
+                (
+                    Node3d::EndMainPass,
+                    Node3d::MotionBlur, // Running before FSR3 reduces edge artifacts and noise
+                    Node3d::Fsr3,
+                    Node3d::Bloom,
+                    Node3d::Tonemapping,
+                ),
+            );
+    }
+}
+
+/// Component to apply FSR3 temporal upscaling and anti-aliasing to a 3D camera.
+///
+/// FSR3 is AMD's temporal upscaling solution that renders at a lower resolution
+/// and upscales to the target resolution using temporal data.
+///
+/// # Tradeoffs
+///
+/// Pros:
+/// * Much better performance by rendering at lower resolution
+/// * High quality temporal anti-aliasing
+/// * Works on AMD, NVIDIA, and Intel GPUs
+/// * Includes optional sharpening pass
```

Comment:
Does the optional sharpening pass make sense to include? We already have RCAS built into Bevy.

I'd rather people use that, as it's compatible with TAA/DLSS, and will play better once we make a generic TSR abstraction.